File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        if 0 in nums:
            if nums.count(0) > 1:
                return [0] * len(nums)

            prod = 0
            other_prod = 1
            for num in nums:
                if num != 0:
                    other_prod = other_prod * num
            
            return [0 if num != 0 else other_prod for num in nums]
        
        else:
            prod = 1
            for num in nums:
                prod = prod * num

            return [int(prod/num) for num in nums]    

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s(nums.begin(), nums.end());
        int longest = 0;

        for (int x: s) {
            if (!s.count(x-1)) {
                int current = x;
                int temp = 1;
                while (true) {
                    current++;
                    if (s.count(current)) {
                        temp++;
                    }
                    else {
                        break;
                    }
                }
            if (temp > longest) {
                longest = temp;
            }
            
            }
        }
        return longest;
    }
};
